app/src/main/java/com/alexbar10/cryptotrack/ui/cryptoAvailable/CryptocurrenciesAvailableFragment.kt

```diff
@@ -0,0 +1,107 @@
+import com.alexbar10.cryptotrack.domain.Cryptocurrency
+import com.alexbar10.cryptotrack.networking.NetworkStatusChecker
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class CryptocurrenciesAvailableFragment : Fragment() {
+
+    private var _binding: FragmentCryptocurrenciesAvailableBinding? = null
+    private val binding: FragmentCryptocurrenciesAvailableBinding
+        get() = _binding!!
+    private lateinit var navController: NavController
+    private val viewModel: CryptocurrenciesAvailableViewModel by viewModels()
+    private lateinit var cryptocurrenciesAdapter: CryptocurrenciesAdapter
+    private val networkStatusChecker by lazy {
+        NetworkStatusChecker(requireActivity().getSystemService(ConnectivityManager::class.java))
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        // Inflate the layout for this fragment
+        return FragmentCryptocurrenciesAvailableBinding
+            .inflate(layoutInflater, container, false)
+            .apply { _binding = this }
+            .root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        cryptocurrenciesAdapter = CryptocurrenciesAdapter { cryptocurrencySelected ->
+            CryptocurrenciesAvailableFragmentDirections
+                .actionCryptocurrenciesAvailableFragmentToCryptocurrencyDetailsFragment(cryptocurrencySelected)
+                .let { navController.navigate(it) }
+        }
+        navController = findNavController()
+        (requireActivity() as MainActivity).supportActionBar?.title = getString(R.string.app_name)
+
+        binding.cryptocurrenciesRecyclerView.run {
+            adapter = cryptocurrenciesAdapter
+            layoutManager = GridLayoutManager(context, 2)
+        }
+        binding.marketMxnCheckbox.setOnCheckedChangeListener { _, isChecked -> viewModel.showCryptocurrenciesFor("mxn", isChecked) }
+        binding.marketArgCheckbox.setOnCheckedChangeListener { _, isChecked -> viewModel.showCryptocurrenciesFor("ars", isChecked) }
+        binding.marketBitcoinCheckbox.setOnCheckedChangeListener { _, isChecked -> viewModel.showCryptocurrenciesFor("btc", isChecked) }
+        binding.marketBrlCheckbox.setOnCheckedChangeListener { _, isChecked -> viewModel.showCryptocurrenciesFor("brl", isChecked) }
+        binding.marketDaiCheckbox.setOnCheckedChangeListener { _, isChecked -> viewModel.showCryptocurrenciesFor("dai", isChecked) }
+        binding.marketUsdCheckbox.setOnCheckedChangeListener { _, isChecked -> viewModel.showCryptocurrenciesFor("usd", isChecked) }
+
+        // Check internet connection
+        networkStatusChecker.performIfConnectedToInternet(
+            {
+                viewModel.getLocalCryptos()
+                Toast.makeText(requireContext(), "Please Your network is unavailable. Check your data or wifi connection", Toast.LENGTH_LONG).show()
```

Comment:
The Toast message should be a String resource to facilitate future changes and potential translations.